docs/source/checks/nlp/model_evaluation/plot_train_test_performance.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+"""
+.. _nlp__train_test_performance:
+
+Train Test Performance for NLP Models
+**************************************
+This notebook provides an overview for using and understanding train test performance check.
+
+**Structure:**
+
+* `What is the purpose of the check? <#what-is-the-purpose-of-the-check>`__
+* `Generate data & predictions <#generate-data-predictions>`__
+* `Run the check <#run-the-check>`__
+* `Define a condition <#define-a-condition>`__
+* `Using a custom scorer <#using-a-custom-scorer>`__
+
+What is the purpose of the check?
+==================================
+This check helps you compare your NLP model's performance between the train and test datasets based on multiple metrics.
+
+For Text Classification tasks the supported metrics are sklearn scorers. You may use any of the existing sklearn
+scorers or create your own. For more information about the supported sklearn scorers, defining your own metrics and
+to learn how to use metrics for other supported task types, see the
+:doc:`Metrics Guide </user-guide/general/metrics_guide>`.
+
+The default scorers that are used for are F1, Precision, and Recall for Classification,
+and F1 Macro, Recall Macro and Precision Macro for Token Classification. See more about the supported task types at the
+:doc:`Supported Tasks </user-guide/nlp/supported_tasks>` guide.
```

Comment:
@Nadav-Barak , @noamzbr  - Ensure this PR is merged after the docs restructure... (?)
(so I guess also the link for supported tasks should be updated...)